core/clustersmngr/factory.go

```diff
@@ -631,5 +641,22 @@ func (cf *clustersManager) getOrCreateClient(ctx context.Context, user *auth.Use
 
 	cf.usersClients.Set(user, cluster.GetName(), client)
 
+	log.V(logger.LogLevelDebug).Info("Client created and added to cache")
+
 	return client, nil
 }
+
+// Format the namespaces as a map[clusterName][]namespacesNames
```

Comment:
Will this scale?

What if you have 100+ namespaces? 1000+ ?

For clarity, 1000+ is not uncommon

Reply:
Good question. For the scenarios we've been exploring ~10-15 namespaces its been a real help. But for 1000 I guess some part of your logging system might get grumpy.

```
{ "myCluster": { "namespaces": ["foo", "bar"], "totalCount": "3" } # show a sensible N namespaces and report the total count
```

I don't know if the structured logging has nested structures in mind..

The count alone would still be useful.